### PR TITLE
Feature/ssl opt

### PR DESCRIPTION
**Pull request by rejoshed**
*Issue #, if available:* TBD

*Description of changes:*
Added the option to set the verify-ssl boolean found on the ACS client. See [here](https://pkg.go.dev/github.com/xanzy/go-cloudstack@v2.4.1+incompatible/cloudstack#NewAsyncClient)

*Note:* This does not enforce the use of an https based protocol. So for the verification to matter, the ACS instance needs to be served at an https endpoint.

The new input config can be written as:
``` ini
[Global]
api-key    = [apiKey] 
secret-key = [secretKey]
api-url    = [https://someAddr:8080/client/api]
verify-ssl = [false|true]
```

And `verify-ssl` is optional, but defaults to true.

*Testing performed:*
Connected via the manager insecurely via http and attempted securely via https protocol. I don't yet have an https version of ACS to test against though.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,9 @@
 bin
 testbin/*
 
+pkg/mocks/*
```
Comment (rejoshed) [inline .gitignore:10]: I also removed the generated mocks from the package.

File: pkg/cloud/client.go [hunk 2]
```diff
@@ -45,37 +46,37 @@ type client struct {
 	// csA *cloudstack.CloudStackClient
 }
 
+// cloud-config ini structure.
+type config struct {
```
Comment (rejoshed) [inline pkg/cloud/client.go:50]: Turns out there's a perfectly natural way to use the `ini` lib that I wasn't using. Derp.

File: pkg/cloud/client.go [hunk 3]
```diff
@@ -45,37 +46,37 @@ type client struct {
...
+	c.cs = cloudstack.NewAsyncClient(cfg.ApiUrl, cfg.ApiKey, cfg.SecretKey, cfg.VerifySSL)
 	_, err = c.cs.Zone.ListZones(c.cs.Zone.NewListZonesParams())
 	if err != nil && strings.Contains(err.Error(), "i/o timeout") {
 		return c, errors.Wrap(err, "Timeout while checking CloudStack API Client connectivity.")
 	}
 	return c, errors.Wrap(err, "Error encountered while checking CloudStack API Client connectivity.")
 }
 
-// CloudStack API config reader.
-func readAPIConfig(cc_path string) (string, string, string, error) {
```
Comment (rejoshed) [inline pkg/cloud/client.go:67]: Just not needed when using the `ini` lib properly.

File: Makefile [hunk 4]
```diff
@@ -170,7 +170,7 @@ test: generate-mocks lint generate-deepcopy bin/ginkgo bin/kubectl bin/kube-apis
 		./hack/testing_ginkgo_recover_statements.sh --remove; exit $$EXIT_STATUS
 	
 .PHONY: generate-mocks
-generate-mocks: bin/mockgen $(shell find ./pkg/mocks -type f -name "mock*.go") ## Generate mocks needed for testing. Primarily mocks of the cloud package.
+generate-mocks: bin/mockgen pkg/mocks/mock_client.go $(shell find ./pkg/mocks -type f -name "mock*.go") ## Generate mocks needed for testing. Primarily mocks of the cloud package.
```
Comment (wongni) [inline Makefile:173]: pkg/mocks/mock_client.go should be returned from $(shell find ./pkg/mocks -type f -name "mock*.go")
Is this additional dependency really required?

File: pkg/cloud/client_test.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud_test
+
+// import (
```
Comment (wongni) [inline pkg/cloud/client_test.go:19]: Why don't you remove this file as this file is technically empty?

Comment (rejoshed) [reply pkg/cloud/client_test.go:19]: Whoops. Didn't mean for it to be commented out like that.

Was trying to figure out why the suite was hanging. It's fixed now.

File: Makefile (hunk 4 above)
Comment (rejoshed) [reply Makefile:173]: It's not found from that if the file doesn't exist to begin with.

File: pkg/cloud/client.go (hunk 2 above)
Comment (jweite-amazon) [reply pkg/cloud/client.go:49]: Good to leverage this.
Looking down the road toward multi-endpoint, I envisioned having a section-per-endpoint, with a user-provided logical name for each section (by which that endpoint would be referred to).  Does the ini lib provide any means for enumerating the sections in an ini file if they're not known a-priori?

**Review by wongni — APPROVED** [review]

**Pull request merged**